packages/store-sync/src/createStoreSync.ts

```diff
@@ -271,25 +272,30 @@ export async function createStoreSync<config extends StoreConfig = StoreConfig>(
       // We use `mergeMap` instead of `concatMap` here to send the fetch request immediately when a new block range appears,
       // instead of sending the next request only when the previous one completed.
       mergeMap(async (blocks) => {
-        const txs = blocks.flatMap((block) => block.logs.map((op) => op.transactionHash).filter(isDefined));
-        if (txs.includes(tx)) return true;
+        for (const block of blocks) {
+          const txs = block.logs.map((op) => op.transactionHash);
+          // If the transaction caused a log, it must have succeeded
+          if (txs.includes(tx)) return { blockNumber: block.blockNumber, status: "success" as const };
+        }
 
         try {
           const lastBlock = blocks[0];
           debug("fetching tx receipt for block", lastBlock.blockNumber);
-          const receipt = await publicClient.getTransactionReceipt({ hash: tx });
-          return lastBlock.blockNumber >= receipt.blockNumber;
+          const { status, blockNumber } = await publicClient.getTransactionReceipt({ hash: tx });
+          if (lastBlock.blockNumber >= blockNumber) {
+            return { status, blockNumber };
```

Comment:
Why not return everything from `getTransactionReceipt`? Then this is like a wrapper for that function

Reply:
because we don't have access to most of this information in the early return case above and it feels wrong to only include this information in some cases based on a race condition